ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/RequiredPluginsInitializer.java

```diff
@@ -35,26 +41,64 @@ public class RequiredPluginsInitializer extends ClasspathContainerInitializer {
 				}
 			});
 
+	private static final DeferredClasspathContainerInitializerJob deferredClasspathContainerInitializerJob = new DeferredClasspathContainerInitializerJob();
+
+	private static class DeferredClasspathContainerInitializerJob extends Job {
+
+		private final Set<IJavaProject> projects = new LinkedHashSet<>();
+
+		public DeferredClasspathContainerInitializerJob() {
+			// This name is not displayed to a user.
+			super("DeferredClasspathContainerInitializerJob"); //$NON-NLS-1$
+			setSystem(true);
+		}
+
+		public synchronized void initialize(IJavaProject project) {
+			if (projects.add(project)) {
+				schedule(5000);
```

Comment:
@laeubi  @iloveeclipse 

Maybe this should use a system property with a default of 5000 so that someone one a slow machine could maybe configure this differently?